lms/djangoapps/grades/new/course_data.py

```diff
@@ -20,23 +25,17 @@ def __init__(self, user, course=None, collected_block_structure=None, course_str
     @property
     def course_key(self):
         if not self._course_key:
-            if self._course:
-                self._course_key = self._course.id
-            else:
-                structure = self._structure or self._collected_block_structure
-                self._course_key = structure.root_block_usage_key.course_key
+            self._course_key = self.course.id
```

Comment:
There was a reason for the original implementation here - I should have added comments.
Essentially, all of these methods are written with performance in mind.  We do not want to access the modulestore to get the course when not needed - so we lazily compute it.  And, if the block_structure is provided, we'd rather use it to retrieve the data - than to make a MongoDB query, instantiate the course module, and retrieve the data from it.
So, in this case, I would want to revert this change.